### PR TITLE
Initial checkin of PullRequest builder code

### PR DESCRIPTION
**Pull request by AdamBrousseau**
- Clone Github repo
- Checkout merge commit of PullRequest
- Clone Eclipse repo, staging branch
- Copy content on disk from Github to Eclipse
- Create commit based on diff
- Push to staging branch
- Staging website http://staging.eclipse.org/openj9/

Part of Issue #12

Signed-off-by: Adam Brousseau <adam.brousseau88@gmail.com>

### CODE REVIEW

**Comment by AdamBrousseau** [comment]
jenkins website stage

**Comment by AdamBrousseau** [comment]
Added some updates to the README. Please review @DanHeidinga

File: jenkinsPullRequests [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+                        git branch: BRANCH, url: ECLIPSE_REPO
+                    }
+                }
+            }
+        }
+        
+        stage('Stage') {
+            steps {
+                timestamps {
+                    dir('staging_repo') {
```
Comment (DanHeidinga) [inline jenkinsPullRequests]: Can you add a comment here to describe what's happening?  It's not immediately obvious that this copies the PR branch from GH over top of the staging branch in the Eclipse git.

File: README.md [hunk 2]
```diff
@@ -10,12 +31,23 @@ Changes made to this repository are automatically promoted.
```
Comment (DanHeidinga) [inline README.md]: Can we rewrite this sentence to say:
`Changes merged into this repositories master branch are automatically synced into the Eclipse git website repo's master branch.  The two master branches should always have the same content, though possibly ~15 minutes delayed due to sync timing`

File: jenkinsPullRequests [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+                        sh 'rm -rf * .gitignore'
+                        sh "cp -r ${WORKSPACE}/github_repo/* ."
+                        sh "cp -r ${WORKSPACE}/github_repo/.gitignore ."
+                        sh 'git status'
+                        sh 'git add -A'
+                        sh 'git status'
+                        sh 'git commit -m "Generated from commit: ${sha1}"'
+                        sh 'git status'
+                        sh "git push origin ${BRANCH}"
+                        sh 'git diff origin/staging..origin/master'
```
Comment (DanHeidinga) [inline jenkinsPullRequests]: Wouldn't it be clearer to add the Eclipse GIT as a new remote and use `git push -f` to write the PR branch over the staging branch?

And so I'm clear, doing so would be equivalent to what you've done here, right?

**Comment by AdamBrousseau** [comment]
jenkins website stage

**Comment by AdamBrousseau** [comment]
jenkins website stage

File: jenkinsPullRequests (hunk 3 above)
Comment (AdamBrousseau) [reply jenkinsPullRequests]: We don't have rights to force push to the Eclipse repo (either master or staging)

**Comment by AdamBrousseau** [comment]
updated commits based on @DanHeidinga's suggestions

File: jenkinsPullRequests (hunk 3 above)
Comment (DanHeidinga) [reply jenkinsPullRequests]: Ok.  Good info to add to a comment in this file to explain why this isn't the idiomatic expression the reader would expect.

File: jenkinsPullRequests (hunk 1 above)
Comment (DanHeidinga) [reply jenkinsPullRequests]: Did this get addressed?

**Comment by DanHeidinga** [comment]
@AdamBrousseau I don't see any comments in the `jenkinsPullRequests` describing what's happening.  Did you forget to push a commit?

**Comment by AdamBrousseau** [comment]
ok, should be all there now

**Pull request merged**